### PR TITLE
tiflash: support session and global variable for FastScan instead of ddl

### PR DESCRIPTION
**Pull request by hongyunyan**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
1. add the usage of session/global variable called 'tiflash_fastscan'.
2. delete the related description about the ddl 'alter table set tiflash mode xx'

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://pingcap.feishu.cn/docx/doxcnUs2ZogSDdH8qDn3KNzLaBd

### Do your changes match any of the following descriptions?

- [x] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-use-fastscan.md [hunk 1]
```diff
@@ -13,34 +13,54 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
 默认情况下，TiFlash 能够保证查询结果精度以及数据一致性。如果使用 FastScan，TiFlash 可以实现更高效的查询性能，但不保证查询结果精度和数据一致性。
 
-某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以将对应的 TiFlash 表开启 FastScan 功能。
+某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以在对应 session 会话中开启 FastScan 功能，你可以通过修改变量```tiflash_fastscan```的值来选择是否启用 FastScan 功能。
 
-对于通过 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md) 启用了 FastScan 的表，FastScan 会全局生效。对于临时表、内存表、系统表、以及列名中含有非 UTF-8 字符的表，都不支持 TiFlash 相关的操作。
+## 启用 FastScan
 
-了解更多信息，请参考 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md)。
+默认情况下，session 和 global 级别的变量 ```tiflash_fastscan=OFF```,即没有开启 FastScan 功能, 你可以通过以下语句来查看对应的变量信息。
```
Comment (joey-yez) [inline develop/dev-guide-use-fastscan.md:20]:
```suggestion
默认情况下，session 和 global 级别的变量 ```tiflash_fastscan=OFF```，即没有开启 FastScan 功能, 你可以通过以下语句来查看对应的变量信息。
```

File: develop/dev-guide-use-fastscan.md [hunk 2]
```diff
@@ -13,34 +13,54 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
-启用完成后，后续对应的表在 TiFlash 中的查询，都会使用 FastScan 功能。
+```
+set global tiflash_fastscan=ON;
+```
 
 可以用下面语句禁用 FastScan。
 
 ```sql
-ALTER TABLE table_name SET TIFLASH MODE NORMAL
+set session tiflash_fastscan=OFF;
```
Comment (joey-yez) [inline develop/dev-guide-use-fastscan.md:63]: 启用 fastscan 介绍了 session 和 global 两种，禁用 fastscan 也应该有两种。

File: develop/dev-guide-use-fastscan.md (hunk 1 above)
Comment (joey-yez) [inline develop/dev-guide-use-fastscan.md]: 这里是否应该使用中文逗号？

**Review by joey-yez — APPROVED** [review]
LGTM

File: develop/dev-guide-use-fastscan.md [hunk 3]
```diff
@@ -13,34 +13,55 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
 默认情况下，TiFlash 能够保证查询结果精度以及数据一致性。如果使用 FastScan，TiFlash 可以实现更高效的查询性能，但不保证查询结果精度和数据一致性。
 
-某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以将对应的 TiFlash 表开启 FastScan 功能。
+某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以在对应 session 会话中开启 FastScan 功能，你可以通过修改变量```tiflash_fastscan```的值来选择是否启用 FastScan 功能。
 
-对于通过 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md) 启用了 FastScan 的表，FastScan 会全局生效。对于临时表、内存表、系统表、以及列名中含有非 UTF-8 字符的表，都不支持 TiFlash 相关的操作。
+## 启用 FastScan
 
-了解更多信息，请参考 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md)。
+默认情况下，session 和 global 级别的变量 ```tiflash_fastscan=OFF```，即没有开启 FastScan 功能，你可以通过以下语句来查看对应的变量信息。
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:20]:
```suggestion
默认情况下，session 和 global 级别的变量 `tiflash_fastscan=OFF`，即没有开启 FastScan 功能，你可以通过以下语句来查看对应的变量信息。
```

File: develop/dev-guide-use-fastscan.md [hunk 4]
```diff
@@ -13,34 +13,55 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
+```
++------------------+-------+
+| Variable_name    | Value |
++------------------+-------+
+| tiflash_fastscan | OFF   |
++------------------+-------+
+```
 
-{{< copyable "sql" >}}
+变量```tiflash_fastscan```支持 session 级别和 global 级别的修改，如果需要在当前 session 中启用 FastScan 功能，可以通过以下语句来设置:
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:48]:
```suggestion
变量 `tiflash_fastscan` 支持 session 级别和 global 级别的修改。如果需要在当前 session 中启用 FastScan 功能，可以通过以下语句来设置:
```

File: develop/dev-guide-use-fastscan.md [hunk 5]
```diff
@@ -13,34 +13,55 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
 
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
 ```
+set session tiflash_fastscan=ON;
+```
+
+也可以对 global 级别的 ```tiflash_fastscan``` 进行设置, 则设置后新建的会话中默认 session 和 global 变量```tiflash_fastscan``` 启用新值。
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:54]:
```suggestion
也可以对 global 级别的 `tiflash_fastscan` 进行设置，则设置后新建的会话中默认 session 和 global 变量 `tiflash_fastscan` 启用新值。
```

File: develop/dev-guide-use-fastscan.md [hunk 6]
```diff
@@ -13,41 +13,62 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
+set session tiflash_fastscan=OFF;
+set global tiflash_fastscan=OFF;
 ```
 
 ## 实现机制
 
 TiFlash 存储层的数据主要存放在 Delta 层和 Stable 层。
 
-Normal Mode 中 TableScan 算子过程整体包括了以下步骤:
+在普通状态下(即未开启 FastScan 功能下) TableScan 算子过程整体包括了以下步骤:
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:71]:
```suggestion
在普通状态下（即未开启 FastScan 功能），TableScan 算子过程整体包括了以下步骤：
```

File: develop/dev-guide-use-fastscan.md [hunk 7]
```diff
@@ -13,41 +13,62 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
 默认情况下，TiFlash 能够保证查询结果精度以及数据一致性。如果使用 FastScan，TiFlash 可以实现更高效的查询性能，但不保证查询结果精度和数据一致性。
 
-某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以将对应的 TiFlash 表开启 FastScan 功能。
+某些 OLAP 对查询结果精度可以容忍一定误差。如果对查询性能有更高要求，可以在对应 session 会话中开启 FastScan 功能，你可以通过修改变量 `tiflash_fastscan` 的值来选择是否启用 FastScan 功能。
 
-对于通过 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md) 启用了 FastScan 的表，FastScan 会全局生效。对于临时表、内存表、系统表、以及列名中含有非 UTF-8 字符的表，都不支持 TiFlash 相关的操作。
+## 启用和禁用 FastScan
 
-了解更多信息，请参考 [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md)。
+默认情况下，session 和 global 级别的变量 `tiflash_fastscan=OFF`，即没有开启 FastScan 功能。你可以通过以下语句来查看对应的变量信息。
```
Comment (shichun-0415) [inline develop/dev-guide-use-fastscan.md:20]: @hongyunyan @hfxsd Seems the variable `tiflash_fastscan` is not documented yet. Please add it to [系统变量](https://docs.pingcap.com/zh/tidb/stable/system-variables).

File: system-variables.md [hunk 8]
```diff
@@ -2309,6 +2309,12 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+- 作用域：SESSION | GLOBAL
```
Comment (hfxsd) [inline system-variables.md:2313]:
```suggestion

- 作用域：SESSION | GLOBAL
```

File: system-variables.md [hunk 9]
```diff
@@ -2309,6 +2309,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`false`
+- 范围：`true | false`
+- 如果开启 FastScan 功能(设置为 true 时)，TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```
Comment (hfxsd) [inline system-variables.md:2317]:
```suggestion
- 如果开启 FastScan 功能（设置为 `true` 时），TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```

File: develop/dev-guide-use-fastscan.md [hunk 10]
```diff
@@ -13,41 +13,62 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
+
+变量 `tiflash_fastscan` 支持 session 级别和 global 级别的修改。如果需要在当前 session 中启用 FastScan 功能，可以通过以下语句来设置:
 
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
 ```
+set session tiflash_fastscan=ON;
+```
+
+也可以对 global 级别的 `tiflash_fastscan` 进行设置，则设置后新建的会话中默认 session 和 global 变量 `tiflash_fastscan` 启用新值。
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:54]:
```suggestion
也可以对 global 级别的 `tiflash_fastscan` 进行设置，设置后新建的会话中默认 session 和 global 变量 `tiflash_fastscan` 启用新值。
```

File: develop/dev-guide-use-fastscan.md [hunk 11]
```diff
@@ -13,41 +13,62 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
+set session tiflash_fastscan=OFF;
+set global tiflash_fastscan=OFF;
 ```
 
 ## 实现机制
 
 TiFlash 存储层的数据主要存放在 Delta 层和 Stable 层。
 
-Normal Mode 中 TableScan 算子过程整体包括了以下步骤:
+在普通状态下（即未开启 FastScan 功能），TableScan 算子过程整体包括了以下步骤：
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:71]:
```suggestion
在默认状态下（即未开启 FastScan 功能），TableScan 算子过程整体包括了以下步骤：
```

File: develop/dev-guide-use-fastscan.md [hunk 12]
```diff
@@ -13,41 +13,60 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
+```
++------------------+-------+
+| Variable_name    | Value |
++------------------+-------+
+| tiflash_fastscan | OFF   |
++------------------+-------+
+```
 
-{{< copyable "sql" >}}
+变量 `tiflash_fastscan` 支持 session 级别和 global 级别的修改。如果需要在当前 session 中启用 FastScan 功能，可以通过以下语句来设置:
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:46]:
```suggestion
变量 `tiflash_fastscan` 支持 session 级别和 global 级别的修改。如果需要在当前 session 中启用 FastScan 功能，可以通过以下语句设置:
```

File: develop/dev-guide-use-fastscan.md [hunk 13]
```diff
@@ -13,41 +13,60 @@ summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
...
 
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+```
+set session tiflash_fastscan=ON;
 ```
 
-启用完成后，后续对应的表在 TiFlash 中的查询，都会使用 FastScan 功能。
+也可以对 global 级别的 `tiflash_fastscan` 进行设置，设置后新建的会话中默认 session 和 global 变量 `tiflash_fastscan` 启用新值。
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:52]:
```suggestion
如果对 global 级别的 `tiflash_fastscan` 进行设置，设置后新建的会话中默认 session 和 global 变量 `tiflash_fastscan` 启用新值。设置方式如下：
```

File: system-variables.md [hunk 14]
```diff
@@ -2309,6 +2309,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`false`
+- 范围：`true | false`
```
Comment (qiancai) [inline system-variables.md:2316]: 参考 develop/dev-guide-use-fastscan.md，是否可以设置为 ON 或 OFF

File: develop/dev-guide-use-fastscan.md [hunk 15]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Fast Mode
-summary: 介绍通过使用 Fast Mode 来加速 OLAP 场景的查询的方法。
+title: FastScan
+summary: 介绍通过使用 FastScan 来加速 OLAP 场景的查询的方法。
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:3]:
```suggestion
summary: 介绍通过使用 FastScan 来加速 OLAP 场景的查询的方法。
aliases: ['/zh/tidb/dev/sql-statement-set-tiflash-mode']

```

File: system-variables.md (hunk 14 above)
Comment (hongyunyan) [reply system-variables.md:2316]: 可以的， on/off ｜ true/false 都是可以的

Comment (hfxsd) [inline system-variables.md:2316]:
```suggestion
- 默认值：`OFF`
- 范围：`ON| OFF`
```

File: system-variables.md [hunk 16]
```diff
@@ -2309,6 +2309,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 范围：`ON| OFF`
+- 如果开启 FastScan 功能（设置为 `true` 时），TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```
Comment (qiancai) [inline system-variables.md:2317]: -
```suggestion
- 如果开启 FastScan 功能（设置为 `ON` 时），TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```

File: system-variables.md [hunk 17]
```diff
@@ -2309,6 +2309,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 范围：`ON| OFF`
```
Comment (qiancai) [inline system-variables.md:2316]:
```suggestion
- 范围：`ON | OFF`
```

**Review by qiancai — APPROVED** [review]

File: system-variables.md [hunk 18]
```diff
@@ -2309,6 +2309,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tiflash_fastscan` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：`OFF`
+- 范围：`ON | OFF`
+- 如果开启 FastScan 功能（设置为 `ON` 时），TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```
Comment (hfxsd) [inline system-variables.md:2317]:
```suggestion
- 如果开启 [FastScan 功能](/develop/dev-guide-use-fastscan.md)（设置为 `ON` 时），TiFlash 可以提供更高效的查询性能，但不保证查询结果的精度和数据一致性。
```

**Review by hfxsd — APPROVED** [review]
LGTM

**Pull request merged**